Wikipedia/Code/DiffContainerViewController.swift

```diff
@@ -0,0 +1,495 @@
+            scrollingEmptyViewController?.view.isHidden = false
+            diffListViewController?.view.isHidden = true
+        case .data:
+            fakeProgressController.stop()
+            scrollingEmptyViewController?.view.isHidden = true
+            diffListViewController?.view.isHidden = false
+            wmf_hideEmptyView()
+            break
+        }
+    }
+    
+    func fetchEditCountIfNeeded() {
+        switch type {
+        case .single:
+            if let username = toModel.user {
+                diffController.fetchEditCount(guiUser: username, siteURL: siteURL) { [weak self] (result) in
+                    
+                    guard let self = self else {
+                        return
+                    }
+                    
+                    DispatchQueue.main.async {
+                        switch result {
+                        case .success(let editCount):
+                            self.updateHeaderWithEditCount(editCount)
+                        case .failure:
+                            break
+                        }
+                    }
+                }
+            }
+        case .compare:
+            break
+        }
+    }
+    
+    func fetchIntermediateCountIfNeeded() {
+        switch type {
+        case .compare:
+            if let fromModel = fromModel {
+                let fromID = fromModel.revisionID
+                let toID = toModel.revisionID
+                diffController.fetchIntermediateCounts(fromRevisionId: fromID, toRevisionId: toID) { [weak self] (result) in
+                    
+                    guard let self = self else {
+                        return
+                    }
+                    
+                    DispatchQueue.main.async {
+                        switch result {
+                        case .success(let counts):
+                            self.updateHeaderWithIntermediateCounts(counts)
+                        case .failure:
+                            break
+                        }
+                    }
+                }
+            } else {
+                assertionFailure("Expect compare type to have fromModel for fetching intermediate count")
+            }
+        case .single:
+            break
+        }
+    }
+    
+    func updateHeaderWithIntermediateCounts(_ counts: (revision: Int, user: Int)) {
+        
+        //update view model
+        let headerViewModel = containerViewModel.headerViewModel
+        
+        switch type {
+        case .compare(let articleTitle):
+            
+            let newTitleViewModel = DiffHeaderViewModel.generateTitleViewModelForCompare(articleTitle: articleTitle, counts: counts)
+            headerViewModel.title = newTitleViewModel
+            headerTitleView?.update(newTitleViewModel)
+        case .single:
+            assertionFailure("Should not call this method for the compare type.")
+            return
```

Comment:
I think this `return` might be redundant

Reply:
agreed, removed